mmtrack/models/track_heads/mask2former_head.py

```diff
@@ -0,0 +1,704 @@
+            a layer to change the embed_dim of transformer encoder in
+            pixel decoder to the embed_dim of transformer decoder.
+            Defaults to False.
+        transformer_decoder (:obj:`ConfigDict` or dict): Config for
+            transformer decoder.
+        positional_encoding (:obj:`ConfigDict` or dict): Config for
+            transformer decoder position encoding.
+            Defaults to `SinePositionalEncoding3D`.
+        loss_cls (:obj:`ConfigDict` or dict): Config of the classification
+            loss. Defaults to `CrossEntropyLoss`.
+        loss_mask (:obj:`ConfigDict` or dict): Config of the mask loss.
+            Defaults to 'CrossEntropyLoss'.
+        loss_dice (:obj:`ConfigDict` or dict): Config of the dice loss.
+            Defaults to 'DiceLoss'.
+        train_cfg (:obj:`ConfigDict` or dict, optional): Training config of
+            Mask2Former head. Defaults to None.
+        test_cfg (:obj:`ConfigDict` or dict, optional): Testing config of
+            Mask2Former head. Defaults to None.
+        init_cfg (:obj:`ConfigDict` or dict or list[:obj:`ConfigDict` or \
+            dict], optional): Initialization config dict. Defaults to None.
+    """
+
+    def __init__(self,
+                 in_channels: List[int],
+                 feat_channels: int,
+                 out_channels: int,
+                 num_classes: int,
+                 num_frames: int,
+                 num_queries: int = 100,
+                 num_transformer_feat_level: int = 3,
+                 pixel_decoder: ConfigType = ...,
+                 enforce_decoder_input_project: bool = False,
+                 transformer_decoder: ConfigType = ...,
+                 positional_encoding: ConfigType = dict(
+                     type='SinePositionalEncoding3D',
+                     num_feats=128,
+                     normalize=True),
+                 loss_cls: ConfigType = dict(
+                     type='CrossEntropyLoss',
+                     use_sigmoid=False,
+                     loss_weight=2.0,
+                     reduction='mean',
+                     class_weight=[1.0] * 133 + [0.1]),
+                 loss_mask: ConfigType = dict(
+                     type='CrossEntropyLoss',
+                     use_sigmoid=True,
+                     reduction='mean',
+                     loss_weight=5.0),
+                 loss_dice: ConfigType = dict(
+                     type='DiceLoss',
+                     use_sigmoid=True,
+                     activate=True,
+                     reduction='mean',
+                     naive_dice=True,
+                     eps=1.0,
+                     loss_weight=5.0),
+                 train_cfg: OptConfigType = None,
+                 test_cfg: OptConfigType = None,
+                 init_cfg: OptMultiConfig = None,
+                 **kwargs) -> None:
+        super(AnchorFreeHead, self).__init__(init_cfg=init_cfg)
```

Comment:
It is ok to change to `super(MMDET_MaskFormerHead, self).__init__(init_cfg=init_cfg)`?

Reply:
I think it should not work. If we inherit from the `Mask2FormerHead`, we needs to provide additional parameters, `num_classes` and `in_channels`, and different type of `in_channels` may also cause errors(`int` in `AnchorFreeHead` vs `List[int]` in `MaskFormerHead`)?